pkg/telemetryquerylanguage/tql/README.md

```diff
@@ -110,18 +110,46 @@ Booleans can be either:
 
-Operators determine how the two Values are compared.  The valid operators are:
+Operators determine how the two Values are compared. 
 
-- Equal (`==`). Equal (`==`) checks if the left and right Values are equal, using Go's `==` operator.
-- Not Equal (`!=`).  Not Equal (`!=`) checks if the left and right Values are not equal, using Go's `!=` operator.
+The valid operators are:
+
+- Equal (`==`). Tests if the left and right Values are equal (see the Comparison Rules below).
+- Not Equal (`!=`).  Tests if the left and right Values are not equal.
+- Less Than (`<`). Tests if left is less than right. 
+- Greater Than (`>`). Tests if left is greater than right.
+- Less Than or Equal To (`<=`). Tests if left is less than or equal to right. 
+- Greater Than (`>`). Tests if left is greater than right.
+
+### Comparison Rules
+
+The table below describes what happens when two Values are compared. Value types are provided by the user of TQL. All of the value types supported by TQL are listed in this table.
+
+If numeric values are of different types, they are compared as `float64`.
+
+For numeric values and strings, the comparison rules are those implemented by Go. Numeric values are done with signed comparisons. For binary values, `false` is considered to be less than `true`. 
+
+For values that are not one of the basic primitive types, the only valid comparisons are Equal and Not Equal, which are implemented using Go's standard `==` and `!=` operators.
+
+A `not equal` notation in the table below means that the "!=" operator returns true, but any other operator returns false.
+
+
+| base type | bool        | int64               | float64             | string                          | Bytes                    | nil                    |
+| --------- | ----------- | ------------------- | ------------------- | ------------------------------- | ------------------------ | ---------------------- |
+| bool      | normal, T>F | not equal           | not equal           | not equal                       | not equal                | not equal              |
+| int64     | not equal   | compared as largest | compared as float64 | not equal                       | not equal                | not equal              |
+| float64   | not equal   | compared as float64 | compared as largest | not equal                       | not equal                | not equal              |
+| string    | not equal   | not equal           | not equal           | normal (compared as Go strings) | not equal                | not equal              |
+| Bytes     | not equal   | not equal           | not equal           | not equal                       | byte-for-byte comparison | not equal              |
+| nil       | not equal   | not equal           | not equal           | not equal                       | not equal                | true for equality only |
```

Comment:
I think it is worth mentioning here that comparing nil with a nil byte slice will return true for equality.